CONTRIBUTING.md

```diff
@@ -48,15 +48,25 @@ The current maintainers are listed in the [CODEOWNERS](.github/CODEOWNERS) file.
 
 If you are interested in being pinged when testing is needed for a certain platform, please add yourself to the [Testers and Contributors](https://github.com/rust-windowing/winit/wiki/Testers-and-Contributors) table!
 
-## Making a new release
-
-If you believe a new release is warranted, you can make a pull-request with:
-- An updated version number (remember to change the version everywhere it is used).
-- A new section in the changelog (below the `# Unreleased` section).
-
-This gives contributors an opportunity to squeeze in an extra PR or two that they feel is valuable
-enough to warrant blocking the release a little.
-
-Once the PR is merged, a maintainer will create a new tag matching the version name (e.g. `v0.26.1`),
-and a CI job will automatically release the new version. Remember that the release date in the
-changelog must be kept in check with the actual release date.
+## Release process
+
+Given that winit is a widely used library we should be able to make a patch
+releases at any time we want without blocking the development of new features.
+
+To achieve these goals, a new branch is created for every new release. Releases
+and later patch releases are committed and tagged in this branch.
+
+The exact steps for an exemplary `0.2.0` release might look like this:
+  1. Initially the version on the latest master is `0.1.0`
```

Comment:
I think the `-dev` suffix is useful to indicate that this is a different version from `0.1.0` and you won't ever find the `0.1.0` version on master.

Reply:
The issue is we want to make it possible to use `patch` section, so if we bump to dev it would mean it's a breaking change and it'll prevent `patch` section from working.

Like for alacritty it makes sense, but we're an application. But for library which you sometimes want to have as breaking that could help?

Reply:
Yeah I guess that makes some sense, though ultimately any git change *could* be a breaking change but that's every commit.

Reply:
Yeah, but you can still can pull and probably fix them somehow. But if you have `-dev` you can't even pull.

Reply:
I mean the version would be updated after release, right? So a breaking commit would then not work with patch anymore either.

But that point is kinda moot because with `-dev` this won't even work for patch versions and once you select that breaking change by picking either `-dev` or major version, you'll have to lock it to a commit anyway unless you want everything to explode constantly.

Reply:
>But that point is kinda moot because with -dev this won't even work for patch versions and once you select that breaking change by picking either -dev or major version, you'll have to lock it to a commit anyway unless you want everything to explode constantly.

I think the issue we're talking about that if you have winit as reexport from e.g. glutin if you bump to `dev` you won't be able to do any updates (even if you intended to fix them) because cargo will prevent it. The point is to allow picking master due to semver compat and fixing issues on your own.